### PR TITLE
IA refresh homepage hero

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/4834

This updates the homepage template to use the same `get_banner` that banner pages do, with effectively a one-liner in the HomePage model to yield the hero banner when that function gets called.

### CODE REVIEW

File: network-api/networkapi/settings.py [hunk 1]
```diff
@@ -242,6 +242,8 @@
     'whitenoise.middleware.WhiteNoiseMiddleware',
     'django.middleware.gzip.GZipMiddleware',
     'networkapi.utility.middleware.TargetDomainRedirectMiddleware',
+    'networkapi.utility.middleware.HTMLCleanUpMiddleware' if DEBUG else None,
```
Comment (Pomax) [inline network-api/networkapi/settings.py:245]: I don't want to add this to prod, as it'll slow things down, but a source prettifier is quite useful. I can spin this off as a separate PR if we prefer, @mmmavis @cadecairos @KalobTaulien

File: network-api/networkapi/wagtailpages/pagemodels/base.py [hunk 2]
```diff
@@ -459,6 +459,9 @@ class Homepage(FoundationMetadataPageMixin, Page):
         related_name='hero_image'
     )
 
+    def get_banner(self):
+        return self.hero_image
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/base.py:463]: allows `{% with banner=page.specific.get_banner %}` in templates to work.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
+
+
   <div class="container">
     <div class="row cutout-wrapper-row">
-      <div class="cutout col-12 col-lg-8 pt-4">
-        <h1 class="h1-heading mb-0">{% if root.title %}{{ root.title }}{% elif page.header %}{{ page.header }}{% else %}{{ page.title }}{% endif %}</h1>
-        {% if page.intro %}
-        <p class="body-large mt-3 mb-0">{{ page.intro }}</p>
-        {% endif %}
+      <div class="cutout col-12 pt-4">
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:36]: this PR is not for the cutout, but these changes are here mostly so I can see things.

File: source/sass/global.scss [hunk 4]
```diff
@@ -38,6 +38,21 @@ hr {
     width: 100%;
     overflow: hidden;
 
+    &.homepage {
+      $bg-pattern-height: 400px;
```
Comment (Pomax) [inline source/sass/global.scss:42]: @mmmavis this is really more of an SCSS stub than anything, to get a rough idea before we actually write the real SCSS =)

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html (hunk 3 above)
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:36]: Let's remove changes made to the cutout section since this affects other pages as well (e.g., hero section of primary page is broken due to this). Without the help of Percy it's hard for us to capture UI changes. It's better that we keep hero and cutout updates in separate PRs.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html [hunk 5]
```diff
@@ -1,34 +1,41 @@
...
-      <source media="(min-width: 768px)" srcset="{{ image_md.url }}" alt="">
-      <source media="(min-width: 576px)" srcset="{{ image_sm.url }}" alt="">
-      {# Fallback Image #}
-      {% image banner fill-1008x288 alt="" %}
-    </picture>
-    {% else %}
-    <img src="{% static "_images/banner-black-white-marble.jpg" %}" alt="">
+  <div class="banner{% if homepage %} homepage{% endif %}">
+    {% if homepage %}
+      <div class="content dark-theme">
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:6]: (Not sure if front end code needs review in this PR... is the code just for placeholder purpose only for now?)

If we want the layout to reflect more like the mockup in this PR - we should add the class `container` here.

File: source/sass/global.scss [hunk 6]
```diff
@@ -38,6 +38,22 @@ hr {
...
+      height: $bg-pattern-height;
+
+      .content {
+        position: relative;
+        z-index: 1;
+      }
+
+      .cutout-wrapper-row {
```
Comment (mmmavis) [inline source/sass/global.scss:50]: Let's revert these changes too.

File: network-api/networkapi/settings.py (hunk 1 above)
Comment (mmmavis) [reply network-api/networkapi/settings.py:245]: I vote for keeping them separate!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html (hunk 5 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/primary_heroguts.html:6]: all front-end work in this PR is purely "stub" work, it's not meant to be even remotely correct, just there to make sure there is a place to pick up from. I'll take it out of this PR if we want to do this once all the backend changes have landed.

**Comment by Pomax** [comment]
@mavis I've split off the html-formatting into a separate PR, and removed the template/css tweaks that I added to be able to see things. That said, I kept a bit in the primary heroguts fragment as placeholder for frontend work.
